src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -787,28 +839,47 @@ private void replaceEC2Servers() {
             // initialize.
             String serverName = String.format("%s %s (%s) %d %s", deployment.r5 ? "r5" : "otp", deployment.name, dateString, serverCounter++, graphAlreadyBuilt ? "clone" : "builder");
             LOG.info("Creating tags for new EC2 instance {}", serverName);
-            ec2.createTags(new CreateTagsRequest()
-                    .withTags(new Tag("Name", serverName))
-                    .withTags(new Tag("projectId", deployment.projectId))
-                    .withTags(new Tag("deploymentId", deployment.id))
-                    .withTags(new Tag("jobId", this.jobId))
-                    .withTags(new Tag("serverId", otpServer.id))
-                    .withTags(new Tag("routerId", getRouterId()))
-                    .withTags(new Tag("user", retrieveEmail()))
-                    .withResources(instance.getInstanceId())
-            );
+            try {
+                getEC2ClientForDeployJob().createTags(new CreateTagsRequest()
+                        .withTags(new Tag("Name", serverName))
+                        .withTags(new Tag("projectId", deployment.projectId))
+                        .withTags(new Tag("deploymentId", deployment.id))
+                        .withTags(new Tag("jobId", this.jobId))
+                        .withTags(new Tag("serverId", otpServer.id))
+                        .withTags(new Tag("routerId", getRouterId()))
+                        .withTags(new Tag("user", retrieveEmail()))
+                        .withResources(instance.getInstanceId())
+                );
+            } catch (Exception e) {
+                status.fail("Failed to create tags for instances.", e);
+                return instances;
```

Comment:
Do the instances need to be terminated here?

Reply:
No, there is a check for job failure after this method in the 2 places it is called.